libraries/ms-drivers/inc/bts7xxx_common.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+// Common definitions for BTS7XXX-series load switches.
+// Most of the functions in this file allow for a common API to easily interact with the
+// EN/SEL pins on the load switches, abstracting the specific method oif accessing the pin.
+
+#include "gpio.h"
+#include "pca9539r_gpio_expander.h"
+#include "soft_timer.h"
+
+#define STM32_GPIO_STATE_SELECT_OUT_0 GPIO_STATE_LOW
+#define STM32_GPIO_STATE_SELECT_OUT_1 GPIO_STATE_HIGH
+#define PCA9539R_GPIO_STATE_SELECT_OUT_0 PCA9539R_GPIO_STATE_LOW
+#define PCA9539R_GPIO_STATE_SELECT_OUT_1 PCA9539R_GPIO_STATE_HIGH
+
+// Represents whether SEL/EN pins are accessed through STM32 or a Pca9539R
+typedef enum {
+  BTS7XXX_PIN_STM32 = 0,
+  BTS7XXX_PIN_PCA9539R,
+  NUM_BTS7XXX_PIN_TYPES,
+} Bts7xxxPinType;
+
+// Holds pin-specific info for EN pins
+typedef struct {
+  GpioAddress *enable_pin_stm32;
+  Pca9539rGpioAddress *enable_pin_pca9539r;
+  SoftTimerId fault_timer_id;
+  bool fault_in_progress;
+  Bts7xxxPinType pin_type;
+} Bts7xxxEnablePin;
+
+// Holds pin-specific info for SEL pins
+typedef struct {
+  GpioAddress *select_pin_stm32;
+  Pca9539rGpioAddress *select_pin_pca9539r;
+  Bts7xxxPinType pin_type;
+} Bts7xxxSelectPin;
+
+// Broad function to enable the pin passed in.
+StatusCode bts7xxx_enable_pin(Bts7xxxEnablePin *pin);

```

Comment:
I think it would be an interesting idea to roll the `fault_in_progress` handling into this function, it looks doable. Then in power distro (and other future projects with a mix of bts7200 and bts7040), to enable a pin, we can just store its `Bts7xxxEnablePin *` and call `bts7xxx_enable_pin` with it instead of worrying about whether a given pin is on a bts7200 or bts7040.

I get the feeling that this module was meant to be private to the bts7200 and bts7040 drivers, though my suggestion would make this function part of the public API of the drivers. You might consider then splitting this module into a `bts7xxx_common` which is public and `bts7xxx_common_impl` which holds private code-sharing functions. I do think `bts7xxx_enable_pin`, `bts7xxx_disable_pin`, and `bts7xxx_get_pin_enabled` at least should be public. Any functions that are part of the public API should be unit tested (functions that are private to bts7200/7040 will/should be tested by those modules' tests and so should be fine without unit tests of their own).